components/google_sheets/sources/new-worksheet/new-worksheet.js

```diff
@@ -0,0 +1,49 @@
+const common = require("../common");
+
+module.exports = {
+  ...common,
+  key: "google_sheets-new-worksheet",
+  name: "New Worksheet (Instant)",
+  description:
+    "Emits an event each time a new worksheet is created in a spreadsheet.",
+  version: "0.0.1",
+  dedupe: "unique",
+  props: {
+    ...common.props,
+    sheetID: {
+      propDefinition: [
+        common.props.google_sheets,
+        "sheetID",
+        (c) => ({
+          driveId: c.watchedDrive === "myDrive" ? null : c.watchedDrive,
+        }),
+      ],
+    },
+  },
+  hooks: {
+    ...common.hooks,
+    async deploy() {
+      await this.processSpreadsheet({ id: this.sheetID });
+    },
+  },
+  methods: {
+    ...common.methods,
+    generateMeta(worksheet) {
+      return {
+        id: worksheet.properties.sheetId,
+        summary: worksheet.properties.title,
+        ts: Date.now(),
+      };
+    },
+    async processSpreadsheet({ spreadsheetId }) {
+      const { sheets: worksheets } = await this.google_sheets.getSpreadsheet(
+        spreadsheetId
+      );
+      for (const worksheet of worksheets) {
+        const meta = this.generateMeta(worksheet);
+        this.$emit(worksheet, meta);
+      }
+    },
```

Comment:
Wouldn't this emit repeated events? The `dedupe` strategy is limited and it's purpose is to avoid duplicate events that happen very close to each other.
Is there a way we can retrieve the creation timestamp of a worksheet? Maybe we can use that to figure out which worksheets are new (similar to the logic for new files).

Reply:
Unfortunately, there's no way to retrieve a creation timestamp for a worksheet. Updated to keep a list of all worksheetIds previously emitted and check against the list to prevent duplicates.